.vib/apache/goss/apache.yaml

```diff
@@ -0,0 +1,49 @@
+file:
+  # Main Apache config file includes other config files
+  {{ .Vars.root_dir }}/apache/conf/httpd.conf:
+    exists: true
+    filetype: file
+    contains:
+      - /^Include.*{{ .Vars.root_dir }}/apache/conf/bitnami/bitnami.conf/
+      - /^Include.*{{ .Vars.root_dir }}/apache/conf/vhosts/\*.conf/
+      - /^Include.*{{ .Vars.root_dir }}/apache/conf/deflate.conf/
+  # Main Bitnami config file was correctly generated
+  {{ .Vars.root_dir }}/apache/conf/bitnami/bitnami.conf:
+    exists: true
+    filetype: file
+    contains:
+      - /DocumentRoot.*{{ .Vars.root_dir }}/apache/htdocs/
+      - /^Include.*{{ .Vars.root_dir }}/apache/conf/bitnami/bitnami-ssl.conf/
+  # Main Bitnami ssl config file was correctly generated
+  {{ .Vars.root_dir }}/apache/conf/bitnami/bitnami-ssl.conf:
+    exists: true
+    filetype: file
+    contains:
+      - "SSLProtocol all -SSLv2 -SSLv3"
+      - /SSLCertificateFile.*bitnami/certs/server.crt/
+      - /SSLCertificateKeyFile.*bitnami/certs/server.key/
+  # CGI folder has been removed, as recommended by Apache
+  {{ .Vars.root_dir }}/apache/cgi-bin:
+    exists: false
+  # Compiled disabled modules should have been generated
+  {{ range $module := .Vars.modules.extra }}
+  {{ $.Vars.root_dir }}/apache/modules/mod_{{ $module }}.so:
+    exists: true
+    filetype: file
+  {{ end }}
+command:
+  # There are no syntax error in the configuration
+  check-configuration:
+    exec: apachectl -t
+    exit-status: 0
+  # Check explicitly enabled and disabled modules
+  check-loaded-modules:
+    exec: apachectl -M
+    exit-status: 0
+    stdout:
+    {{ range $module := .Vars.modules.loaded }}
+      - "{{ $module }}_module"
+    {{ end }}
```

Comment:
Same than above. Is it necessary to check all modules here? I'd check only the most used ones (proxy, rewrite, ssl)

Reply:
Well this I understand it’s a dynamic testing. I will keep it as it is and if we need to change anything is in the vars file but this code is fine IMHO

Reply:
The code is dynamic as mentioned and adapts to whatever values in `vars.yaml`. 

> Is it necessary to check all modules here?

Regarding the list of checked modules, we are not testing all modules here (there are over 100 modules in the container):

```bash
$ ls -l /opt/bitnami/apache/modules/ | wc -l
98
```

We are only checking the ones meeting these criteria:

- We are [explicitly enabling them in postunpack](https://github.com/bitnami/containers/blob/a88a8046c4f4e0b2c6452e2dca435a818da01b08/bitnami/apache/2.4/debian-11/rootfs/opt/bitnami/scripts/apache/postunpack.sh#L27)
- We are [explicitly disabling them in postunpack](https://github.com/bitnami/containers/blob/a88a8046c4f4e0b2c6452e2dca435a818da01b08/bitnami/apache/2.4/debian-11/rootfs/opt/bitnami/scripts/apache/postunpack.sh#L44)
- The module was compiled because we set explicit flags during compilation for it